src/test/java/org/prebid/server/bidder/somoaudience/SomoaudienceBidderTest.java

```diff
@@ -94,17 +95,19 @@ public void makeHttpRequestShouldReturnErrorMessageWhenMediaTypeIsAudio() {
         final BidRequest bidRequest = BidRequest.builder()
                 .imp(singletonList(Imp.builder()
                         .id("impId")
+                        .banner(Banner.builder().build())
                         .audio(Audio.builder().build())
-                        .ext(Json.mapper.valueToTree(ExtPrebid.of(null, ExtImpSomoaudience.of("placementId")))).build()))
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpSomoaudience.of("placementId", null)))).build()))
                 .build();
 
         // when
         final Result<List<HttpRequest<BidRequest>>> result = somoaudienceBidder.makeHttpRequests(bidRequest);
 
         // then
-        assertThat(result.getErrors()).hasSize(1)
-                .containsExactly(BidderError.badInput(
-                        "ignoring imp id=impId, Somoaudience doesn't support Audio"));
+        assertThat(result.getErrors()).hasSize(2)
+                .containsExactlyInAnyOrder(BidderError.badInput(
+                        "ignoring imp id=impId, Somoaudience doesn't support Audio"),
+                        BidderError.badInput("No valid impressions"));
         assertThat(result.getValue()).isEmpty();
     }
 
```

Comment:
Please add a test with input request containing two imps with banners and different placement hashes, one video and one native imp. In `then` section please verify correctness of produced request body.